gaiku-common/src/data/chunk.rs

```diff
@@ -1,97 +1,125 @@
 use mint::{Vector3, Vector4};
 
-// TODO: Get inspiration on multiarray crate (https://github.com/sellibitze/multiarray) to make chunk 2d and 3d friendly
+pub trait Chunkify {
+  fn new(position: [f32; 3], width: usize, height: usize, depth: usize) -> Self;
```

Comment:
I realize you have already merged this but thought I'd comment here rather than make a new issue. But do we need a `new` in the `Chunkify` trait surely this should be a user land thing. For example I have no need for `position` in my code my dimensions are usually specified by a `Vector3<usize>`. Is there anywhere in the library where this is used and therefore must be in the `Chunkify` trait rather than just standard `Chunk` struct?

Reply:
Maybe I should move the initialization method to another trait? If you check the lastest update in #14 you will see that I changed the baker and file format to use generics structs that implements the traits, so one solution could be to provide a default implementation (with unimplemented! macro) to this constructor so it is not required to be implemented by you but if you use it the app will panic, and maybe change the name from `new` to `with_position_and_size` so it doesn't get in your way if you want to define your own initialization method.

Reply:
The issue I have with `with_position_and_size` is that it is not really generic like the other trait methods. Is there a way we could construct it generically? For example I have a chunk which has a tree that grows, the data is sparse since most of the tree is empty space so rather than a pre-init arrays it uses a `HashMap<(usize,usize,usize), u8>` which is set only when there is something there. At baking I iter the map to find out the stats like dimensions.

What about `from_data(data: HashMap<(usize,usize,usize),(u8,Vector4<u8>))>` the file format loader can create a map instead of interacting with the chunk directly then load it all at once into a new chunk letting the user land decide if they want fixed sizes or dynamic ones. I would love it if we could do something like this too:

```rust
let chunk: Chunk = data.into();
```

But this is still not ideal as it would probably be slower. Hmmm designing libraries is hard haha

Reply:
For now we should probably just go with a seperate trait and `with_position_and_size` can also work up another PR with imporments another day